### PR TITLE
ADD: useAsyncPromise

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

File: class/useAsyncPromise.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+import { useState, useEffect } from 'react';
+
+/**
+ * A custom React hook that accepts a promise and returns the resolved value and any errors that occur.
+ *
+ * @template T - The type of the resolved value.
+ * @param {Promise<T>} promise - The promise to be resolved.
+ * @returns {{ data: T | null, error: Error | null, loading: boolean }} - An object with the resolved data, any error, and loading state.
+ */
+function useAsyncPromise<T>(promise: Promise<T>): { data: T | null; error: Error | null; loading: boolean } {
```
Comment (Overtorment) [inline class/useAsyncPromise.ts:10]: File is added, but where is it used..?

**Review by Overtorment — CHANGES_REQUESTED** [review]

File: class/useAsyncPromise.ts (hunk 1 above)
Comment (marcosrdz) [reply class/useAsyncPromise.ts:10]: on add.js

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/b5m4lrf2x5r6p7st2m26dwxtlu?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11701)

**Review by Overtorment — CHANGES_REQUESTED** [review]
It's not a class tho? Prob better to move it to "hooks" folder

**Comment by marcosrdz** [comment]
> It's not a class tho? Prob better to move it to "hooks" folder

Done

**Review by Overtorment — APPROVED** [review]

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/heg2iykd4cq3otmfypgm457f5q?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11703)

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/prl6jayhaejou6rnc7xkbdtrxq?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/11705)

**Comment by GladosBlueWallet** [comment]
Wake the fuck up samurai, we have PRs to merge

image

[all PRs for @limpbrains] https://github.com/BlueWallet/BlueWallet/pulls/review-requested/limpbrains

File: screen/wallets/add.js [hunk 2]
```diff
@@ -51,7 +52,7 @@ const WalletsAdd = () => {
   const [walletBaseURI, setWalletBaseURI] = useState('');
   const [selectedIndex, setSelectedIndex] = useState(0);
   const [label, setLabel] = useState('');
-  const [isAdvancedOptionsEnabled, setIsAdvancedOptionsEnabled] = useState(false);
+  const isAdvancedOptionsEnabled = useAsyncPromise(isAdvancedModeEnabled());
```
Comment (limpbrains) [inline screen/wallets/add.js:55]: isAdvancedModeEnabled function will be called once again, one component re-renders. I don't think it is expected

**Comment by GladosBlueWallet** [comment]
Unbelievable. You, [subject name here], must be the pride of [subject hometown here]!

**Pull request merged**